linters/amp_fix_md_links.py

```diff
@@ -63,6 +63,38 @@ def _make_path_module_agnostic(path: str) -> str:
     return upd_path
 
 
+def _check_md_section_exists(
+    link_in_cur_module: str,
+    section: str,
+    link_path: str,
+    file_name: str,
+    line_num: int,
+) -> str | None:
```

Comment:
We use Optional[str], so let's do that for consistency but we should switch the code base to the new approach at some point